### DISCUSSION TITLE
Column name conflicts, how to solve them without changing the column name?

### DISCUSSION THREAD

**Discussion by JCKodel** [simolus3/drift#2067] [Q&A]
I'm working on a simple project and, so far, I had two conflicting column names that will not compile because the same property/method exists in the generated database.

The first one was `text` (a column cannot be called text because the database class already has a text() method). Ok, renamed the column to `entry`. Problem averted.

Now, `tableName` is an issue (because tableName is in the generated database code in the same scope). Problem now is that my server requires a column named `tableName` =\

How one can avoid those kind of conflicts?

**Comment by simolus3** [comment]
I'm surprised this is undocumented, but in a `CREATE TABLE` statement from a drift file, you can use an `AS` constraint to specify the getter name of a Dart column without changing it's schema name, e.g.

```sql
CREATE TABLE myTable (
    tableName TEXT AS tableNameColumn -- will be called `tableNameColumn` in Dart
);
```

Of course, for Dart-defined tables, you can just use `TextColumn get tableNameColumn => text().named('tableName')()` if you need to.
